nengo_viz/static/viz_netgraph_conn.js

```diff
@@ -0,0 +1,155 @@
+     *  the connection will look for the next item on the list, and so on
+     *  until it finds one that does exist. */
+    this.pres = info.pre;
+    this.posts = info.post;
+
+    this.removed = false;
+
+    /** the actual NetGraphItem currently connected to/from */
+    this.pre = null;
+    this.post = null;
+
+    /** figure out the best available items to connect to */
+    this.set_pre(this.find_pre());
+    this.set_post(this.find_post());
+
+    /** create the line and its arrowhead marker */
+    this.g = ng.createSVGElement('g');
+
+    this.line = ng.createSVGElement('line');
+    this.g.appendChild(this.line);
+    
+    this.marker = ng.createSVGElement('path');
+    this.marker.setAttribute('d', "M 10 0 L -5 -5 L -5 5 z");
+    this.g.appendChild(this.marker);
+
+    this.redraw();
+
+    ng.g_conns.appendChild(this.g);
+
+}
+
+
+/** set the item connecting from */
+VIZ.NetGraphConnection.prototype.set_pre = function(pre) {
+    if (this.pre != null) {
+        /** if we're currently connected, disconnect */
+        var index = this.pre.conn_out.indexOf(this);
+        if (index == -1) {
+            console.log('error removing in set_pre');
+        }
+        this.pre.conn_out.splice(index, 1);    
+    }
+    this.pre = pre;
+    /** add myself to pre's output connections list */
+    this.pre.conn_out.push(this);
+}
+
+
+/** set the item connecting to */
+VIZ.NetGraphConnection.prototype.set_post = function(post) {
+    if (this.post != null) {
+        /** if we're currently connected, disconnect */
+        var index = this.post.conn_in.indexOf(this);
+        if (index == -1) {
+            console.log('error removing in set_pre');
+        }
+        this.post.conn_in.splice(index, 1);    
+    }
+    this.post = post;
+    /** add myself to post's input connections list */
+    this.post.conn_in.push(this);
+}
+
+
+/** determine the best available item to connect from */
+VIZ.NetGraphConnection.prototype.find_pre = function() {
+    for (var i in this.pres) {
+        var pre = this.ng.svg_objects[this.pres[i]];
+        if (pre != undefined) {
+            return pre;
+        } else {
+            /** register to be notified if a better match occurs */
+            this.ng.register_conn(this, this.pres[i]);
+        }
+    }
+    console.log('could not find pre');
+    console.log(this.pres);
+}
+
+
+/** determine the best available item to connect to */
+VIZ.NetGraphConnection.prototype.find_post = function() {
+    for (var i in this.posts) {
+        var post = this.ng.svg_objects[this.posts[i]];
+        if (post != undefined) {
+            return post;
+        } else {
+            /** register to be notified if a better match occurs */
+            this.ng.register_conn(this, this.posts[i]);
+        }
+    }
+    console.log('could not find post');
+    console.log(this.posts);
+}
+
+
+/** remove this connection */
+VIZ.NetGraphConnection.prototype.remove = function() {
+    if (this.parent != null) {
```

Comment:
Hmm... on second thought, I'm very tempted to stick with a standard that all if conditions should evaluate to booleans, rather than relying on automatic boolean conversion.  As I was going though looking at places where I could replace ```if (x != null)``` with ```if (x)```, I started getting confused as to which ones were actually boolean variables and which ones were doing this auto-type conversion.

That said, I'm definitely open to discussion on this one, but for now I like the explicit approach.